holmes/core/tool_calling_llm.py

```
@@ -997,20 +1067,51 @@ def call_stream(
+            
+            # Extract just the termination decision section
+            if "# Termination Decision Prompt" in full_kaito_content:
+                termination_prompt = full_kaito_content.split("# Termination Decision Prompt\n\n", 1)[1]
+            else:
+                logging.warning("🔍 TERMINATION DEBUG: Template section not found, cannot perform intelligent termination")
+                return False  # Continue if template is missing
+
+            # Create a fresh conversation for termination check
+            termination_messages = [
+                {"role": "system", "content": "You are a helpful assistant that evaluates whether investigations should continue."},
+                {"role": "user", "content": f"Based on this conversation history:\n\n{str(messages)}\n\n{termination_prompt}"}
```

Ruff (0.14.10)

Use explicit conversion flag

Replace with conversion flag

(RUF010)